packages/server/src/api/controllers/screen.ts

```diff
@@ -46,12 +46,19 @@ export async function save(
 
   const isCreation = !screen._id
 
-  if (
-    (await features.isEnabled(FeatureFlag.WORKSPACE_APPS)) &&
-    screen.workspaceAppId &&
-    !(await sdk.workspaceApps.get(screen.workspaceAppId))
-  ) {
-    ctx.throw("workspaceAppId is not valid")
+  if (await features.isEnabled(FeatureFlag.WORKSPACE_APPS)) {
+    if (
+      screen.workspaceAppId &&
+      !(await sdk.workspaceApps.get(screen.workspaceAppId))
+    ) {
+      ctx.throw("workspaceAppId is not valid")
+    } else if (!screen.workspaceAppId) {
+      let [workspaceApp] = await sdk.workspaceApps.fetch()
```

Comment:
```suggestion
      const [workspaceApp] = await sdk.workspaceApps.fetch()
```

Reply:
I realized after suggesting this that it's not correct, but for some reason github won't let me delete this comment so please ignore.